gitops/core.py

```diff
@@ -326,11 +326,15 @@ def unsetenv(ctx, filter, values, exclude=""):
         print(success_negative("Aborted."))
         return
     for app in apps:
-        environment = app.values.get("environment", {})
+        environment = app.values.get("environment") or {}
         for e in splitenvs:
             if e in environment:
                 del environment[e]
-        update_app(app.name, environment=_sort_envs(environment))
+        update_app(
+            app.name,
+            delete_when_empty=False,
```

Comment:
I don't like this -- let's just always delete. Might trigger some unwanted deploys the first time round, but anything touching these files should be doing so hygienically and not leaving behind mess.

Reply:
Yeah, I prefer force normalizing it all too, but if that's the case, we should make that a proper requirement and prevent people from committing manually written yamls that doesn't conform to the standard.

We can pave the desired path here by having a linter in pre-commit/GHA to block committing denormalized yamls (and more importantly, to block commits that would cause superfluous redeployments).

Reply:
If we want to go that way, I can revert all of the behavior change in this PR and just leave the tests for documentation/spec. We'll just cop the redeployments caused by https://github.com/uptick/uptick-cluster/pull/48 for now. If the hand written yamls becomes annoying again, maybe I'll write a linter then.

Do we have an agreement here?

Reply:
Hand-written yamls are annoying but rare. Linter would be nice, but I'm fine with it just getting tidied up next time some other CLI action touches it.

Let's just make sure CLI deletes any empty envs always.
Cop the redeployments in uptick/uptick-cluster#48

Reply:
(by linter, here I actually meant just running `gitops unsetenv all <some-definitely-non-existent-variables>` and checking that it doesn't create any git commit)